cedar-policy-core/src/parser/cst_to_ast.rs

```diff
@@ -3840,11 +3863,24 @@ mod tests {
         let invalid_is_policies = [
             (
                 r#"permit(principal in Group::"friends" is User, action, resource);"#,
-                ExpectedErrorMessageBuilder::error("expected an entity uid or matching template slot, found an `is` expression").exactly_one_underline(r#"Group::"friends" is User"#).build(),
+                ExpectedErrorMessageBuilder::error("when `is` and `in` are used together, `is` must come first")
+                    .help("try `_ is _ in _`")
+                    .exactly_one_underline(r#"principal in Group::"friends" is User"#)
+                    .build(),
+            ),
+            (
+                r#"permit(principal, action in Group::"action_group" is Action, resource);"#,
+                ExpectedErrorMessageBuilder::error("`is` cannot appear in the action scope")
+                    .help("try moving `action is ..` into a `when` condition")
+                    .exactly_one_underline(r#"Group::"action_group" is Action"#)
```

Comment:
This underline is strange, because this text isn't part of the original policy. Does the test pass?

Reply:
The underline is still a little strange because the help text talks about `action is`, but `action` is not part of the underline

Reply:
might be clearer to underline the whole `action` clause?

Reply:
Agreed

Reply:
Done